src/sandmastery/Java/leaf/cosmere/sandmastery/common/utils/MiscHelper.java

```diff
@@ -0,0 +1,101 @@
+package leaf.cosmere.sandmastery.common.utils;
+
+import leaf.cosmere.api.*;
+import leaf.cosmere.api.helpers.StackNBTHelper;
+import leaf.cosmere.api.manifestation.Manifestation;
+import leaf.cosmere.api.spiritweb.ISpiritweb;
+import leaf.cosmere.client.Keybindings;
+import leaf.cosmere.common.cap.entity.SpiritwebCapability;
+import leaf.cosmere.sandmastery.common.registries.SandmasteryBlocksRegistry;
+import leaf.cosmere.sandmastery.common.registries.SandmasteryItems;
+import net.minecraft.core.BlockPos;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.util.Mth;
+import net.minecraft.world.effect.MobEffectInstance;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.phys.AABB;
+import net.minecraftforge.registries.ForgeRegistries;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class MiscHelper {
+    public static boolean checkIfNearbyInvestiture(ServerLevel pLevel, BlockPos pPos) {
+
+        int range = 10;
+        AABB areaOfEffect = new AABB(pPos).inflate(range, range, range);
+        List<LivingEntity> entitiesToCheckForInvesiture = pLevel.getEntitiesOfClass(LivingEntity.class, areaOfEffect, e-> true);
+
+        AtomicBoolean foundSomething = new AtomicBoolean(false);
+        for (LivingEntity targetEntity : entitiesToCheckForInvesiture)
+        {
+			MobEffectInstance copperEffect = targetEntity.getEffect(ForgeRegistries.MOB_EFFECTS.getValue(new ResourceLocation("allomancy", "copper_cloud")));
```

Comment:
We should move the `ForgeRegistries.MOB_EFFECTS.getValue` call outside the loop, so it's not recreating the reference and resource location a bunch of times

Reply:
and if they don't have allomancy installed, this may return null, so would be good to null-guard